### ISSUE TITLE
Default unit tests are failing

### ISSUE THREAD

**Issue by maxpou** [vuejs/vue-cli#1879]
### Version
3.0.0-rc.5

### Reproduction link
[https://codesandbox.io/s/vue](https://codesandbox.io/s/vue)

### Steps to reproduce
```
vue create vue-jest
? Please pick a preset: Manually select features
? Check the features needed for your project: Babel, Router, Vuex, Linter, Unit
? Pick a linter / formatter config: Basic
? Pick additional lint features: Lint on save
? Pick a unit testing solution: Jest
? Where do you prefer placing config for Babel, PostCSS, ESLint, etc.? In dedicated config files
? Save this as a preset for future projects? No
```

duplicate `HelloWorld.spec.js` (i.e. Users.spec.js)

### What is expected?
test should pass.

### What is actually happening?
```
 FAIL  tests/unit/Users.spec.js
  ● Test suite failed to run

    /lab/vue-jest/tests/unit/Users.spec.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import "core-js/modules/es6.array.iterator";
                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    SyntaxError: Unexpected string

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:403:17)
```

---
In jest.config.js I added `cache: false` to solve the error.
Other information:
* yarn v 1.6.0
* node v10.1.0
* macos High Sierra

**Comment by yyx990803** [comment]
Can't reproduce with Yarn 1.7 & Node 10.3.0 on macOS. Try upgrading Yarn?

**Comment by maxpou** [comment]
I'm now with yarn 1.7. Problem persist.
I also tried without yarn: `rm -rf node_modules && npm i && npm run test:unit`. And I have the problem too. (with npm v 6.1.0)

**Comment by javacoldbrew** [comment]
The problem seems to be with the environment variables `VUE_CLI_BABEL_TARGET_NODE` and `VUE_CLI_BABEL_TRANSPILE_MODULES`. If they are set to true, the issue goes away.

[Jianwu Chen](https://stackoverflow.com/users/2817457/jianwu-chen) on stackoverflow.com found this solution:

https://stackoverflow.com/questions/51365250/run-jest-got-unexpected-string-at-scripttransformer-transformandbuildscript

**Comment by yyx990803** [comment]
Are you running `jest` directly? That's not supposed to work. You must run it via `vue-cli-service test:unit`.

**Comment by yyx990803** [comment]
@javacoldbrew FYI just landed cd88cfa1 which allows you to debug vue-cli-service itself via Node inspector.

**Comment by javacoldbrew** [comment]
@yyx990803 I feel silly, I am pretty sure that was my problem. I was running the tests through IntelliJ's Jest runner which just calls `jest` directly, which would sometimes work.

I will double check and get back to you tomorrow and see if I still get the error when using `yarn run test:unit`.

Anyways, thank you for all your help!

**Comment by javacoldbrew** [comment]
@yyx990803 I feel silly, I am pretty sure that was my problem. I was running the tests through IntelliJ's Jest runner which just calls `jest` directly, which would sometimes work.

I will double check and get back to you tomorrow and see if I still get the error when using `yarn run test:unit`.

Anyways, thank you for all your help!

**Comment by maxpou** [comment]
@yyx990803 I'm using the `yarn run test:unit` / `npm run test:unit`

**Comment by mghignet** [comment]
Hi,

What version of `babel-jest` are you using?
I had the same issue with version `23.0.1` and I solved it using version `23.4.0`.
I made a mistake when trying to fix versions in my `package.json` which led me to use the wrong one.
If that can help... 👍

**Comment by erwin16** [comment]
upgrading to 23.4.0 raises a lot of wrnings;;;

		 yarn upgrade babel-jest@23.4.0
		yarn upgrade v1.9.2
		[1/4] 🔍  Resolving packages...
		[2/4] 🚚  Fetching packages...
		[3/4] 🔗  Linking dependencies...
		warning "@vue/cli-plugin-babel > babel-loader@8.0.0-beta.4" has unmet peer dependency "webpack@>=2".
		warning "@vue/cli-plugin-eslint > eslint-loader@2.1.0" has unmet peer dependency "webpack@>=2.0.0 <5.0.0".
		warning "@vue/cli-plugin-pwa > workbox-webpack-plugin@3.4.1" has unmet peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
		warning "@vue/eslint-config-prettier > eslint-config-prettier@2.9.0" has unmet peer dependency "eslint@>=3.14.1".
		warning " > babel-core@7.0.0-bridge.0" has unmet peer dependency "@babel/core@^7.0.0-0".
		warning " > sass-loader@7.0.3" has unmet peer dependency "webpack@^3.0.0 || ^4.0.0".

**Comment by ntsim** [comment]
Updating `jest` to 23.4.2 seems to cause these failing tests for me. There appears to be some dependencies that don't play well together. I could not quite track down the true source. Even after trawling through Jest's recent changes, it does not look like anything specifically could cause this.

Anyone else running into this issue, you should be able to workaround it by installing `jest-cli` and pinning it exactly to 23.4.1.

**Comment by ChristianAEDev** [comment]
I believe I solved this problem by adding the following to the `jest.config.js`:

```js
transformIgnorePatterns: [
    "node_modules/(?!(babel-jest|jest-vue-preprocessor)/)"
  ]
```
I sadly have to say *"I believe..."*, because since I added this option to a newly created Vue project (using *Vue CLI v3.0.0-rc.10*) it suddenly also works for another project, where I didn't add this option.

I am therefore not exactly sure what solved the issue and I am not able to reproduce it anymore.
But I wanted to leave this here for others to try out.

The following links helped me to find this solution:
* https://medium.com/@kevin_peters/unit-testing-vue-js-components-with-jest-86e14ef499da
In this article an adjustment to the *package.json* was made to ensure proper pre-processing of *.vue* files.
* https://jestjs.io/docs/en/tutorial-react-native.html#transformignorepatterns-customization
It describes what the `transformIgnorePatterns` option does and how to set it up for React Native.

**Comment by simonjohansson** [comment]
@ChristianAEDev 

Thanks, this solved it for me, until I change a test file. After running `./node_modules/jest/bin/jest.js --clearCache` the tests are working again.

```
$ cat tests/unit/HelloWorld.spec.ts
import { shallowMount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';

describe('HelloWorld.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message';
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg },
    });
    expect(wrapper.text()).toMatch(msg);
  });
});

$ npm run test:unit
> frontend@0.1.0 test:unit /Users/sjo4473/src/simonjohansson/quiz/frontend
> vue-cli-service test:unit

 PASS  tests/unit/HelloWorld.spec.ts
  HelloWorld.vue
    ✓ renders props.msg when passed (26ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.303s, estimated 2s
Ran all test suites.

$ vim tests/unit/HelloWorld.spec.ts
$ cat tests/unit/HelloWorld.spec.ts
import { shallowMount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';

describe('HelloWorld.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message';
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg },
    });
    expect(wrapper.text()).toMatch(msg);
  });

  it('blurgh', () => {
    expect(true).toBeTruthy();
  });
});

$ npm run test:unit
> frontend@0.1.0 test:unit /Users/sjo4473/src/simonjohansson/quiz/frontend
> vue-cli-service test:unit

 FAIL  tests/unit/HelloWorld.spec.ts
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /Users/sjo4473/src/simonjohansson/quiz/frontend/tests/unit/HelloWorld.spec.ts:3
    import "core-js/modules/es6.array.iterator";
    ^^^^^^

    SyntaxError: Unexpected token import

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:403:17)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.079s
Ran all test suites.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! frontend@0.1.0 test:unit: `vue-cli-service test:unit`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the frontend@0.1.0 test:unit script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/sjo4473/.npm/_logs/2018-08-02T15_05_54_261Z-debug.log

$ ./node_modules/jest/bin/jest.js --clearCache
Cleared /var/folders/2z/g6z_y_4d6hn03vfx9k9gwfp8bxwvxb/T/jest_6mf3xn

$ npm run test:unit
> frontend@0.1.0 test:unit /Users/sjo4473/src/simonjohansson/quiz/frontend
> vue-cli-service test:unit

 PASS  tests/unit/HelloWorld.spec.ts
  HelloWorld.vue
    ✓ renders props.msg when passed (23ms)
    ✓ blurgh (1ms)

Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        2.163s
Ran all test suites.
```

**Comment by ebertti** [comment]
This is not a perfect soluction, but I think this can be used temporarily

```js
  transformIgnorePatterns: [
    '/node_modules/.*',
  ],
```

It works for me.

**Comment by yyx990803** [comment]
For users seeing the same problem: do you have a default `NODE_ENV` set for your shell?

**Comment by mathieumure** [comment]
@yyx990803, I have the same problem and no default NODE_ENV are set in my shell.

**Comment by yyx990803** [comment]
@mathieumure still happening with RC11?

**Comment by simonjohansson** [comment]
I noticed that as soon as I stopped using the `jest` plugin in Visual Studio Code the issue went away for me.

**Comment by mathieumure** [comment]
@yyx990803 stop happening after stopping using jest directly from both my shell and IDE...I miss read the whole thread, my bad 😅 .
But still, I wonder if using vue-cli-service instead of jest will not broke a lot of integration in IDE...

**Comment by ntsim** [comment]
After spending some time debugging the issue, I think I've come up with a feasible solution. It seems like this comes down to a 'wtf' moment in terms of how NPM's dependency resolution strategy works. The following seems to be the solution for me, but your mileage may vary.

### TLDR

You can do either of these things:

1. Clear down `node_modules` and remove `package-lock.json`. Re-run `npm install`.
2. Run `npm cache verify` and re-run `npm install`. 

I personally think doing both is the best option to ensure everything is sound and your `package-lock.json` is more in-line with Vue CLI when it generates a fresh project.

### Long explanation

The intention seems to be that `cli-plugin-unit-jest` tries to depend on the root level `babel-core`, which in combination with `cli-plugin-babel` is usually `babel-core@7.0.0-bridge.0`. This is normally fine, but for some unlucky people, NPM can incorrectly resolve `babel-core@6.26.3` as a nested dependency of the plugin (instead of being at the root level of `node_modules`). As it's essentially the wrong version, all sorts of weird things probably happen behind the scenes, but the bottom line is that Jest cannot transform the code correctly.

Why does this happen? I'm not 100% sure (I am no expert in the NPM dependency resolution algorithm) but my best guess is that:

- `cli-plugin-unit-jest` relies on `babel-core@6.26.3` indirectly through one of it's dependencies. I think it is `babel-jest` in this case, but as it is a peer dependency, it means that `babel-core@7.0.0-bridge.0` should theoretically be used.

- When NPM has to resolve dependencies for `babel-jest` on the project level e.g. when updating it in my case, the algorithm gets confused and believes `babel-core` can't be resolved as a peer dependency. I believe this is at least partly due to the NPM cache being corrupted or caching incorrect dependencies in some way. NPM resolves `babel-core@6.26.3` as a nested dependency instead and everything breaks.

It seems like the way to resolve this involves ensuring the NPM cache is correct and consistent (i.e. `npm cache verify`), and/or rebuilding the `package-lock.json` and `node_modules` entirely to make sure it resolves everything from scratch again (without any external caching factors).

**Comment by Nivani** [comment]
I found a workaround based on the one posted by @ebertti and the code of the vue-plugin-unit-test module.

Adding the following at the top of the jest.config.js file fixes the problem:
```javascript
process.env.VUE_CLI_BABEL_TARGET_NODE = true;
process.env.VUE_CLI_BABEL_TRANSPILE_MODULES = true;
```

**Comment by uglow** [comment]
Thanks @ntsim - your solution seems to have worked. 

For other people struggling with this problem (which for me looked like this:
```
$npm run test:unit

> web-common@0.1.0 test:unit /code/monorepo/packages/web-common
> vue-cli-service test:unit

 FAIL  src/components/Header.spec.js
  ● Test suite failed to run

    Plugin 0 specified in "/code/monorepo/packages/web-common/node_modules/@vue/babel-preset-app/index.js" provided an invalid property of "default" (While processing preset: "/code/monorepo/packages/web-common/node_modules/@vue/babel-preset-app/index.js")

      at Plugin.init (../../node_modules/jest-config/node_modules/babel-core/lib/transformation/plugin.js:131:13)
          at Array.map (<anonymous>)
          at Array.map (<anonymous>)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        2.123s

```
), I've been using Lerna with sub-packages and that seems to re-produce the error every time I run `lerna bootstrap --hoist`. As soon as I do a straight `npm install` in the sub-package (after having applied your solution), it works again. 

**Update**: It also works if I use `lerna bootstrap` without the `--hoist` option.

**Comment by macdaddyaz** [comment]
> I found a workaround based on the one posted by @ebertti and the code of the vue-plugin-unit-test module.
> 
> Adding the following at the top of the jest.config.js file fixes the problem:
> 
> ```js
> process.env.VUE_CLI_BABEL_TARGET_NODE = true;
> process.env.VUE_CLI_BABEL_TRANSPILE_MODULES = true;
> ```

Personally, I found this solution proposed by @Nivani to be sufficient without any other changes (in a project generated with CLI 3.0.5). I also found that if I don't want to "pollute" the Jest config file, I can just set the env vars in the IDE's run configuration.

Another option is to create a separate config file for the IDE to use that looks like this:

```js
process.env.VUE_CLI_BABEL_TARGET_NODE = true;
process.env.VUE_CLI_BABEL_TRANSPILE_MODULES = true;

const config = require( './jest.config');
module.exports = config;
```

I called it `jetbrains-jest.config.js` and then selected it as the "Configuration file" in the Jest run configuration. That way, Vue CLI can be Vue CLI, and WebStorm can be WebStorm.

**Comment by Aaron-Pool** [comment]
The simple

```
process.env.VUE_CLI_BABEL_TARGET_NODE = true;
process.env.VUE_CLI_BABEL_TRANSPILE_MODULES = true;
```

addition at the top of my jest config file seems to have fixed things for me as well.

**Comment by LinusBorg** [comment]
Weird. `vue-cli-service test:unit` sets these variables for you before running jest:

https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-plugin-unit-jest/index.js#L13-L15

The question then would be why they are lost for some people.

**Comment by cars10** [comment]
I also faced that issue. What helped me was:

* setting `babel-core` to `~7.0.0-bridge.0` in `package.json`
* running `npx jest --clearCache`

After that everything worked again.

**Comment by leumashz** [comment]
@cars10 Solution worked for me.

**Comment by jackkoppa** [comment]
@cars10, also worked for me. After installing `babel-core` at `~7.0.0-bridge.0`, we're using

```json
"test:unit": "jest --clearCache && vue-cli-service test:unit",
```

right now, to avoid the errors whenever we test. Obviously, that's less than ideal if our tests start to get larger. And we're also not super comfortable about still requiring `babel-core` when we should likely only be using 
```json
"@babel/core": "^7.1.5",
```

So I'll keep an eye on this

**Comment by dasDaniel** [comment]
> 
> 
> I found a workaround based on the one posted by @ebertti and the code of the vue-plugin-unit-test module.
> 
> Adding the following at the top of the jest.config.js file fixes the problem:
> 
> ```js
> process.env.VUE_CLI_BABEL_TARGET_NODE = true;
> process.env.VUE_CLI_BABEL_TRANSPILE_MODULES = true;
> ```

this is, IMHO, way faster than waiting for all tests to finish, when you're only working on one file

**Comment by Himujjal** [comment]
> 
> 
> I also faced that issue. What helped me was:
> 
>     * setting `babel-core` to `~7.0.0-bridge.0` in `package.json`
> 
>     * running `npx jest --clearCache`
> 
> 
> After that everything worked again.

This worked like a charm! Thanks

**Comment by plondon** [comment]
@cars10 you just saved my life, if you accept bitcoin ill send you some 💸

**Comment by begueradj** [comment]
I tried every single suggestion provided so far, in vain.

**Comment by araujo0205** [comment]
I Create a new project in vue-cli add "Unit Testing" and copy differences in my project. This work for me.
Install the package:
```
npm install --save-dev @vue/cli-plugin-unit-jest
```
change test:unit in `package.json` to:
```
"test:unit": "vue-cli-service test:unit"
```
And create file `jest.config.js` with this value:
```
module.exports = {
  moduleFileExtensions: [
    'js',
    'jsx',
    'json',
    'vue'
  ],
  transform: {
    '^.+\\.vue$': 'vue-jest',
    '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$': 'jest-transform-stub',
    '^.+\\.jsx?$': 'babel-jest'
  },
  moduleNameMapper: {
    '^@/(.*)$': '<rootDir>/src/$1'
  },
  snapshotSerializers: [
    'jest-serializer-vue'
  ],
  testMatch: [
    '**/tests/unit/**/*.spec.(js|jsx|ts|tsx)|**/__tests__/*.(js|jsx|ts|tsx)'
  ],
  testURL: 'http://localhost/'
}
```
In my test, work without `jest.config.js`.

**Comment by killboard** [comment]
> I Create a new project in vue-cli add "Unit Testing" and copy differences in my project. This work for me.
> Install the package:
> 
> ```
> npm install --save-dev @vue/cli-plugin-unit-jest
> ```
> change test:unit in `package.json` to:
> 
> ```
> "test:unit": "vue-cli-service test:unit"
> ```
> And create file `jest.config.js` with this value:
> 
> ```
> module.exports = {
>   moduleFileExtensions: [
>     'js',
>     'jsx',
>     'json',
>     'vue'
>   ],
>   transform: {
>     '^.+\\.vue$': 'vue-jest',
>     '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$': 'jest-transform-stub',
>     '^.+\\.jsx?$': 'babel-jest'
>   },
>   moduleNameMapper: {
>     '^@/(.*)$': '<rootDir>/src/$1'
>   },
>   snapshotSerializers: [
>     'jest-serializer-vue'
>   ],
>   testMatch: [
>     '**/tests/unit/**/*.spec.(js|jsx|ts|tsx)|**/__tests__/*.(js|jsx|ts|tsx)'
>   ],
>   testURL: 'http://localhost/'
> }
> ```
> In my test, work without `jest.config.js`.

This solution works fine. Please update the guide at
https://vue-test-utils.vuejs.org/guides/testing-single-file-components-with-jest.html

It is not guiding to run the tests with jest through `vue-cli`

**Comment by timcluyts** [comment]
Like the OP says, for me too putting `cache: false` in my `jest.config.js` worked and solved the issue

**Comment by themeler** [comment]
I've encountered same problem after updating `babel-jest` to version `24.1.0`. None of the solutions helped. Getting back to `23.6.0` solved problem for now.

**Comment by rafagarcia** [comment]
@killboard your solution was the one. In my case i followed this steps:

- Uninstall Jest extension from VSC
- npm install @vue/cli-plugin-unit-jest

**Comment by androidfanatic** [comment]
Removing `babel-jest` from package.json worked for me.

`Babel-jest` was set to version `24.7.1` in package.json and package-lock.json. Just removed it, then did  `rm -rf node_modules` followed by `npm i` and tests started working.

**Comment by ktquez** [comment]
Solution
https://github.com/vuejs/vue-cli/issues/1879#issuecomment-409872897

**Comment by gabrieloliverio** [comment]
Had the same problem. I was importing a lib built in esm format and getting `SyntaxError: Unexpected token export`.

I followed the [ChristianAEDev](https://github.com/vuejs/vue-cli/issues/1879#issuecomment-409872897) sugestion, including this lib in the `transformIgnorePatterns` and it worked!

```
transformIgnorePatterns: [
    "node_modules/(?!(library)/)"
  ],
```

**Comment by MaximStone** [comment]
Only that worked for me:

```
module.exports = {
  moduleFileExtensions: ['js', 'jsx', 'json', 'vue'],
  transform: {
    '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$': 'jest-transform-stub',
    **// process js with babel-jest
    '^.+\\.jsx?$': '<rootDir>/node_modules/babel-jest',
    // process *.vue files with vue-jest
    '.*\\.(vue)$': '<rootDir>/node_modules/vue-jest',**
  },
  transformIgnorePatterns: ['/node_modules/.*'],

  moduleNameMapper: {
    '^@/(.*)$': '<rootDir>/src/$1',
  },
  snapshotSerializers: ['jest-serializer-vue'],
  testMatch: ['**/tests/unit/**/*.spec.(js|jsx|ts|tsx)|**/__tests__/*.(js|jsx|ts|tsx)'],
  testURL: 'http://localhost/',
  watchPlugins: ['jest-watch-typeahead/filename', 'jest-watch-typeahead/testname'],
};

```

**Comment by MarioKusek** [comment]
I just deleted `jest.config.js` and now tests are working.

**Comment by MarioKusek** [comment]
I have solved my problem by switching to node LTS (10.16.3) and changing version of babel-jest to ^23.6.0. Now tests work as expected.

**Comment by iliacmd** [comment]
`module.exports = {
  presets: [
    ['@vue/babel-preset-app', { modules: false }],
  ],
  plugins: [
    'dynamic-import-node',
  ],
  env: {
    test: {
      presets: [
        ['@vue/babel-preset-app', { modules: 'commonjs', useBuiltIns: false }],
      ],
    },
  },
};`

**Comment by stalniy** [comment]
Had the same issue. The root cause is if you use `npm` and it doesn't hoist dependencies to the `node_modules` folder. Run `npm ls vue-jest babel-jest jest-transform-stub`, it should show that all deps at the root level:

```
└─┬ @vue/cli-plugin-unit-jest@3.11.0
  ├── babel-jest@23.6.0 
  ├─┬ jest@23.6.0
  │ └─┬ jest-cli@23.6.0
  │   └─┬ jest-config@23.6.0
  │     └── babel-jest@23.6.0  deduped
  ├── jest-transform-stub@2.0.0 
  └── vue-jest@3.0.5 
```

if not, then jest just cannot `require` transformers using `require`. Then you need to change the path to transformer. For example:

```js
// jest.config.js
module.exports = {
// .....
  transform: {
    '^.+\\.vue$': 'vue-jest',
   '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$': 'jest-transform-stub',
    "^.+\\.[t|j]sx?$": "<rootDir>/node_modules/@vue/cli-plugin-unit-jest/node_modules/babel-jest"
  },
}
```
this works in case you have this from `npm ls vue-jest`:

```
└─┬ @vue/cli-plugin-unit-jest@3.11.0
  └── vue-jest@3.0.5 
```

Alternatively run `npm dedupe` to hoist and dedupe dependencies (it will update your package-lock.json as well, so the next time you won't have that issue)

**Comment by rudolfbyker** [comment]
Simply running `jest --clearCache` fixed it for me. I guess this is the same as a lot of the previous solutions that implicitly causes that cache to be cleared or solutions that explicitly disable that cache.

No changes to my project were needed to fix it, nor did anything in the project change when it broke. So on my side, it was not an issue with dependency resolution. I still don't know what confused jest in the first place, but I was installing and uninstalling various packages on the CLI when it happened.

It also looks like not everyone in this thread is having the exact same problem...

**Comment by webxmsj** [comment]
I used the vant component library when I was executing Vue-cli-service test:unit The following error was encountered during the command:
![image](https://user-images.githubusercontent.com/40289489/91851383-a3c17880-ec91-11ea-967d-a08fa01e21c7.png)
How should I solve the problem
